pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -158,27 +166,29 @@ func getPackagesFromCB(ctx context.Context, clusterBootstrap *runtanzuv1alpha3.C
 			if pkgShortName == "tkg-storageclass" {
 				continue
 			}
-			packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: pkgShortName, Namespace: constants.TkgNamespace},
+			packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: pkgShortName, Namespace: systemNamespace},
 				Spec: kapppkgv1alpha1.PackageSpec{RefName: pkgName, Version: pkgVersion}})
 		}
 	}
 
 	return packages, nil
 }
 
-// checkClusterCB checks if clusterbootstrap resource is created correctly and packages are reconciled successfully on cluster
-func checkClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName string, isManagementCluster bool) error {
+// CheckClusterCB checks if clusterbootstrap resource is created correctly and packages are reconciled successfully on cluster
+func CheckClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName string, isManagementCluster bool) error {
 	log.Infof("Verify addons on workload cluster %s with management cluster %s", wcClusterName, mcClusterName)
 
 	var clusterBootstrap *runtanzuv1alpha3.ClusterBootstrap
-	if isManagementCluster {
+
+	// in tkgs infrastructure, CB only exists in the supervisor cluster
+	if infrastructureName == "tkgs" || isManagementCluster {
 		clusterBootstrap = getClusterBootstrap(ctx, mccl, mcClusterNamespace, mcClusterName)
 	} else {
-		clusterBootstrap = getClusterBootstrap(ctx, mccl, wcClusterNamespace, wcClusterName)
+		clusterBootstrap = getClusterBootstrap(ctx, wccl, wcClusterNamespace, wcClusterName)
```

Comment:
This change of client for the workload cluster is wrong. As explained above in the previous comment, CB always reside in the management cluster. You should use ```mccl``` for both as it was before your change.

Reply:
Fixed